hypervisor/arch/x86/vmx.c

```diff
@@ -764,30 +759,10 @@ static void init_host_state(void)
 	exec_vmwrite(VMX_HOST_GDTR_BASE, gdtb.base);
 	pr_dbg("VMX_HOST_GDTR_BASE: 0x%x ", gdtb.base);
 
-	/* TODO: Should guest TR point to host TR ? */
-	trbase = gdtb.base + tr_sel;
-	if (((trbase >> 47U) & 0x1UL) != 0UL) {
-		trbase |= 0xffff000000000000UL;
-	}
-
-	/* SS segment override */
-	asm volatile ("mov %0,%%rax\n"
-		      ".byte 0x36\n"
-		      "movq (%%rax),%%rax\n":"=a" (trbase_lo):"0"(trbase)
-	    );
-	realtrbase = ((trbase_lo >> 16U) & (0x0ffffUL)) |
-	    (((trbase_lo >> 32U) & 0x000000ffUL) << 16U) |
-	    (((trbase_lo >> 56U) & 0xffUL) << 24U);
-
-	/* SS segment override for upper32 bits of base in ia32e mode */
-	asm volatile (
-		      ".byte 0x36\n"
-		      "movq 8(%%rax),%%rax\n":"=a" (trbase_hi):"0"(trbase));
-	realtrbase = realtrbase | (trbase_hi << 32U);
-
-	/* Set up host and guest TR base fields */
-	exec_vmwrite(VMX_HOST_TR_BASE, realtrbase);
-	pr_dbg("VMX_HOST_TR_BASE: 0x%016llx ", realtrbase);
+	tss_addr = (uint64_t)(&get_cpu_var(tss));
```

Comment:
get_cpu_var returns the virtual address, we should convert to physical address before setting VMCS field.